keps/sig-release/20190219-publishing-packages.md

```diff
@@ -0,0 +1,381 @@
+    And they configure their system's package manager to use that key
+    And they use their system's package manager to install a package from this specific kubernetes minor release
+   Then their package manager will not complain about untrusted packages, sources or repositories
+```
+
+<!--
+```
+Scenario: [...]
+  Given ...
+    And ...
+   When ...
+   Then ...
+```
+-->
+
+### Implementation Details/Notes/Constraints
+
+There should be the following options endusers should be able to use as repositories for their systems' package managers:
+
+Packages will be published for different ...
+- **`${dist}`**: a combination of `<distribution>-<code-name-or-version>`
+  (e.g. `debian-jessie`, `ubuntu-xenial`, `fedora-23`, `centos-7`, ...)
+- **`${k8s_release}`**: the version of kubernetes `<major>.<minor>`
+  (e.g. `1.12`, `1.13`, `1.14`, ...)
+- **`${channel}`**: can be `stable`, `dev`, `nightly`
+    - `stable`: all official releases for `${k8s_release}`
+      (e.g.: `1.13.0`, `1.13.1`, `1.13.2`, ...)
+    - `dev`: all development releases for all minor releases in this `${k8s_release}`, including `alpha`s, `beta`s and `rc`s
+      (e.g.: `1.13.0-rc.2`, `1.13.2-beta.0`, `1.13.1-alpha.3`, ...)
+    - `nightly`: any package cut automatically on a daily basis
+
+Therefore a configuration for package managers might look something like:
+
+- deb:
+    ```
+    # deb http://apt.kubernetes.io/${dist} ${k8s_release} ${channel}
+    deb http://apt.kubernetes.io/debian-jessie 1.13 nightly
+    ```
+- rpm/yum:
+    ```
+    [kubernetes]
+    name=Kubernetes
+    # baseurl=http://yum.kubernetes.io/${dist}/${k8s_release}/${channel}
+    baseurl=http://yum.kubernetes.io/fedora-27/1.13/nightly
+    enabled=1
+    gpgcheck=1
+    repo_gpgcheck=1
+    gpgkey=file:///etc/pki/rpm-gpg/kubernetes.gpg.pub
+    ```
+
+Different architectures will be published into the same repos, it is up to the package managers to pull and install the correct package for the target platform.
+
+
+Implementation steps:
+- [ ] get minimal infra in place
+    - [ ] buckets, dns, repos, ...
+- [ ] create additional step in [anago] to generate packages on staging step [via the new method][pkg-gen-kep]
+- [ ] create additional step in [anago] to publish packages to new repos on publish time
+- [ ] get tests in place (see: [Test Plan](#Test-Plan))
+- [ ] adapt official documentation to point to new repositories
+
+> [color=#ff0000] **TODO**
+> - how to manage shared secrets (sigining key), what do we need for that?
+> - which distributions, platforms, ... we support is probably depending on:
+>   - which packages [we can build][pkg-gen-kep]
+>   - how long all those builds will take
+> - probably starting out with current debian, ubuntu, fedora and RHEL versions?
+
+### Risks and Mitigations
+
+> [color=#ff0000] **TODO**
+
+- *Risk*: We don't find a proper way to share secrets like the singing key*
```

Comment:
```suggestion
- *Risk*: We don't find a proper way to share secrets like the signing key*
```

Reply:
This has been fixed already. I cannot even commit this suggestion ...

![image](https://user-images.githubusercontent.com/1401333/53559821-f1ac6c00-3b42-11e9-9da0-d74594e11afd.png)

Will resolve this conversation. However thanks for reviewing ...

Reply:
That was weird 🙃 thanks @hoegaarden 😄